### ISSUE TITLE
Not able to get access to global variable for Vue 3

### ISSUE THREAD

**Issue by darkman97i** [vuejs/vuex#2009]
### Version
4.0.2

### Reproduction link
[https://github.com/darkman97i/kcenter-mobile](https://github.com/darkman97i/kcenter-mobile)

### Steps to reproduce
* In the project I have shared a README.md where I'm trying to explain the issue.
* Not clear for me if it is a bug or wrong configuration from my side. I only get working in half times ( maybe is something with hooks? )

thanks for your time

### What is expected?
standard way to access a global variable from vuex

### What is actually happening?
Not able to access a global variable from vuex 4.0.2 with Vue 3.x. Not found sample or documentation to get working with Vue 3.x

---

    I have other projects working with Vue 2 and I succeed get working but with vuex version 4.x and Vue 3 I do not succeed to get working in all cases.
    If you share some documentation section where explain how doing it, will be perfect, because I do not succeed on find it and I think this should be a standard feature.

**Comment by cuebit** [comment]
`getCurrentInstance` is only available in life cycle hooks, or `setup` with Composition API. Its usage in application code is strongly discouraged.

That said, you may want to avoid component-related logic in Vuex for this very reason. Perhaps instantiate the WebService and access it directly, say, from a module? Or pass `this.$ws` from your components to the Vuex actions that require it? The ball is in your court.

Closing since it's neither a bug nor a feature request.

**Comment by darkman97i** [comment]
Do not hate me for reopening the ticket :) but I would like to do an extra explanation.

* I know it is possible to send from component `this.$ws` to the storage on each action request from the component -> but not much practical when you will have more than 20 modules and a lot of actions into.
* I created only a single Webservice for the whole application because it keeps not the only implementation also some values on the fly like authentication token -> reuse it in all the modules is very useful ( otherwise I have similar problem -> I need authentication token on each module to use the webservice ).

In case of Vue 2 I was able to do something like ( seems in new Vue 3 I should forget this kind of ideas ): 
```
Import Vue from 'vue';

const okm = new Vue();
// at this point I was able to access okm.$ws from vuex
```

If I understood, arrived at this point I'm thinking that a had a bad pattern design and in the vuex storage should not implement logic into -> the best practice should be to implement specific module layer with the logic into at the top from where applying changes in vuex, is that?

Thanks for your patience.
